numpy/core/src/multiarray/lowlevel_strided_loops.c.src

```diff
@@ -1707,10 +1712,22 @@ mapiter_@name@(PyArrayMapIterObject *mit)
                 }
 
                 /*
-                 * Resetting is slow, so skip if the subspace iteration has
-                 * only a single inner loop.
+                 * Resetting is slow, so skip if the subspace iteration is
```

Comment:
Let me try to explain nditer a bit, I think it will make things clear why it should work. I am not sure it might help but...

The advanced iterator will always do some "iteration", after each of these there is an inner loop optimization. This inner loop can be iterated trivially by a single stride. Now in principle the inner stride you need can change of course.
This is why there is the check that the innerloop is actually the complete iteration. If the innerloop spans the whole iteration there is no reason for the innermost stride to change later on (assuming no buffering, buffering might add weirdness). The only reason why this could happen is if the strided layout is different for different "subspaces/inner iteration positions", but that is not compatible to numpy's memory layout.
All in all, even if you change the iterator to do a different iteration order, these things should still work as far as I can tell.